src/main/java/nextstep/courses/infrastructure/JdbcSessionRepository.java

```diff
@@ -0,0 +1,215 @@
+
+    private List<Long> findRegisterUsers(long sessionPk) {
+        String sql = "select user_id from user_session_registration where session_id = ?";
+
+        return jdbcTemplate.query(sql, rs -> {
+            List<Long> idList = new ArrayList<>();
+            while (rs.next()) {
+                idList.add(rs.getLong(1));
+            }
+            return idList;
+        }, sessionPk);
+    }
+
+    @Override
+    public Session findById(Long id) {
+        String sql = "select id, course_id, state, cover_image_file_path, image_type, image_capacity, image_width, image_height, start_time, end_time, fee\n" +
+                "from session\n" +
+                "where id = ?";
+
+        RowMapper<Session> rowMapper = (rs, rowNum) -> {
+            long pk = rs.getLong(1);
+
+            ImageExtension extension = ImageExtension.valueOf(rs.getString(5));
+            MaxRegister maxUserCount = MaxRegister.of(rs.getInt(11));
+            if(rs.wasNull()) {
+                maxUserCount = MaxRegister.infinite();
+            }
+
+            RegisteredUsers registeredUsers = new RegisteredUsers();
+            for (long userPk : findRegisterUsers(pk)) {
+                NsUser user = userRepository.findById(userPk);
+                registeredUsers.add(user);
+            }
+
+            return Session.create(
+                    pk,
+                    courseRepository.findById(rs.getLong(2)),
+                    SessionState.valueOf(rs.getString(3)),
+                    registeredUsers,
+                    new SessionImage(rs.getString(4), rs.getInt(6), rs.getInt(7), rs.getInt(8), extension),
+                    new SessionDuration(toLocalDateTime(rs.getTimestamp(9)), toLocalDateTime(rs.getTimestamp(10))),
+                    maxUserCount,
+                    rs.getInt(11)
+                    );
+        };
+        return jdbcTemplate.queryForObject(sql, rowMapper, id);
+    }
+
+    private LocalDateTime toLocalDateTime(Timestamp timestamp) {
+        if (timestamp == null) {
+            return null;
+        }
+        return timestamp.toLocalDateTime();
+    }
+}
+
+/**
+ * 세션 객체를 DB 레코드로 매핑하는 책임을 가진 객체
+ * 객체를 탐색하며 DB에 매핑될 필드를 찾아냅니다.
+ */
+class RawSession {
+    private final Session original;
+
+    public RawSession(Session original) {
+        this.original = original;
+    }
+
+    public Long id() {
+        return original.getId();
+    }
+
+    public Long courseId() {
+        return original.getCourse().getId();
+    }
+
+    public String state() {
+        return original.getState().toString();
+    }
+
+    public String coverImageFilePath() {
+        return original.getCoverImage().getFilePath();
+    }
+
+    public String imageType() {
+        return original.getCoverImage().getType().toString();
+    }
+
+    public int imageCapacity() {
+        return original.getCoverImage().getCapacity().toInt();
+    }
+
+    public int imageWidth() {
+        return original.getCoverImage().getSize().getWidth();
```

Comment:
```suggestion
        return original.getImageWidth();
```

또는 

```suggestion
        return original.coverImage().width();
```

메서드 이름에 get을 빼고 구현하는 것은 어떨까?